api/docs/feature-development/model.md

```diff
@@ -0,0 +1,348 @@
+
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  # ...
+  describe '.feed_for' do
+    let(:user) { create(:user, profile: build(:profile)) }
+    let(:author) { create(:author, profile: build(:profile)) }
+    let(:relationship) { create(:relationship, follower: user, followed: author) }
+
+    before(:each) do
+      create_list(:article, 3, author: build(:author))
+      create_list(:article, 3, author: author)
+    end
+
+    subject { described_class.feed_for(relationship.follower) }
+
+    it { expect(subject.length).to be 3 }
+  end
+end
+```
+
+Here, we've created a `user`, `author`, `relationship` and 6 `articles`.
+
+Our expectation is that we should only see the 3 `articles`, for the `author` that we followed.
+
+> You might be wondering why we are using the user from the relationship, it's just so we don't need to call `user.reload` after we've created the `relationship`.
+
+Let's make that test pass!
+
+```rb
+# frozen_string_literal: true
+
+class Article < ApplicationRecord
+  # ...
+  def self.feed_for(user)
+    return none unless user.present?
+
+    where(author: user.following)
+  end
+end
+```
+
+Nice and clean, just a simple where clause `author` is a `user` that is being `followed`.
+
+### Favorites count
+
+```rb
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  # ...
+  describe '#favorites_count' do
+    let(:user) { create(:user) }
+    let(:article) { create(:article, author: build(:author)) }
+    let(:favorite) { build(:favorite, user: user, article: article) }
+
+    it 'will increment favorites_count by 1 when user favorites an article' do
+      expect do
+        favorite.save!
+      end .to change { article.favorites_count }.by(1)
+    end
+
+    it 'will decrement favorites_count by 1 when user unfavorites an article' do
+      favorite.save!
+
+      expect do
+        favorite.destroy!
+      end .to change { article.favorites_count }.by(-1)
+    end
+  end
+end
+```
+
+we added a `favorites_count` field to our table, let's test the expection of what should happen to it:
+
+1. When a user favorites an article
+   1. `favorites_count` should be incremented.
+2. When a user unfavorites an article
+   1. `favorites_count` should be decremented.
+
+Let's make the test pass!
+
+```rb
+# frozen_string_literal: true
+
+class Favorite < ApplicationRecord
+  belongs_to :article, counter_cache: true
+end
+```
+
+Rails has built in support for incrementing/decrementing counts, so this is all we need to do!
+
+> You might be wondering, why did we test what rails provides us?
+>
+> Great question! The reason we wrote tests is because its a part of our business logic, these are expectations of what **should** happen, it doesn't matter that rails provides it to us, because if rails changed something, or we wanted to use something else in the future, the test will still stay the logic of what is supposed to happen, will stay the same.
```

Comment:
Great question! The reason we wrote tests is because its a part of our business logic, these are expectations of what **should** happen. Even though the functionality is something that rails provides to us, the tests help us preserve and communicate the business logic expectations.